pkg/config/data.go

```diff
@@ -118,15 +123,64 @@ type File struct {
 	Email            string `toml:"email"`
 	Endpoint         string `toml:"endpoint"`
 	LastVersionCheck string `toml:"last_version_check"`
+
+	// Allow the API endpoint itself to be dynamically switched if we wanted it to
+	APIEndpoint string `toml:"api_endpoint"`
+}
+
+// Load gets the configuration file from the CLI API endpoint and encodes it
+// from memory into config.File.
+func (f *File) Load(configEndpoint string, httpClient api.HTTPClient) error {
+	fmt.Println("We were unable to locate a local configuration file required to use the CLI.")
+	fmt.Println("We will create that file for you now.")
+
+	f.APIEndpoint = configEndpoint
+
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+	defer cancel()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, f.APIEndpoint, nil)
+	if err != nil {
+		return err
+	}
+
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	_, err = toml.DecodeReader(resp.Body, f)
+	if err != nil {
+		return err
+	}
+
+	// Create the destination directory for the config file
+	basePath := strings.Split(FilePath, "config.toml")[0]
```

Comment:
Not sure if this will be OK for non-unix systems 🤷🏻

Reply:
I believe you want [`filepath.Base()`](https://golang.org/pkg/path/filepath/#Base) which should be safe cross-platform

Reply:
Ah nice!

Reply:
@phamann Actually I don't think that is what we want as it gives back `config.toml` instead of the path leading to that section.

Reply:
Ah there we go `filepath.Dir`